src/Core/Checkout/Cart/SalesChannel/CartOrderRoute.php

```diff
@@ -169,4 +166,21 @@ private function addAffiliateTracking(Cart $cart, DataBag $data): void
             $cart->setCampaignCode($campaignCode);
         }
     }
+
+    private function place(Cart $cart, SalesChannelContext $context, RequestDataBag $data): mixed
+    {
+        $calculatedCart = $this->cartCalculator->calculate($cart, $context);
+
+        $response = $this->checkoutGatewayRoute->load(new Request($data->all(), $data->all()), $cart, $context);
+        $calculatedCart->addErrors(...$response->getErrors());
+
+        $this->taxProviderProcessor->process($calculatedCart, $context);
+
+        $this->addCustomerComment($calculatedCart, $data);
+        $this->addAffiliateTracking($calculatedCart, $data);
+
+        Profiler::trace('checkout-order::pre-payment', fn() => $this->paymentProcessor->validate($calculatedCart, $data, $context));
+
+        return Profiler::trace('checkout-order::order-persist', fn() => $this->orderPersister->persist($calculatedCart, $context));
```

Comment:
`$this->orderPersister->persist` returns a string, where comes the `OrderPlaceResult` into place? 

I am not sure, if this code would work right now

Reply:
fixed